src/Addon/Plugin/PluginCriteria.php

```diff
@@ -198,8 +213,8 @@ public function getCacheKey()
             $this->options,
             'cache.key',
             $this->getCachePrefix() . '.'
-            . md5(json_encode($this->collection)) . '.'
```

Comment:
`$this->collection` is nearly static. It hashes the collectionClass name (string). that nearly never changes. So no need to include this.